src/field.ml

```diff
@@ -1,12 +1,18 @@
 include Field_intf
 
-type kind = _kind [@@deriving repr]
+type nonrec kind = kind [@@deriving repr]
 
 module MakeInt (Size : SIZE) = struct
   type t = int [@@deriving repr]
 
   let size = Size.size
 
+  (* Your [Int(32,64).(to,of)_int] uses are unsafe. You should do something about it.
+
+     [get_int32_be b off |> Int32.to_int] would be wrong on large enough ints on 32bit systems
```

Comment:
I think this one is fine ?

Reply:
`int32` has one more bit than `int` on 32 bit systems, but more importantly, `int64` has one more bit than `int` on 64 bit systems.

Reply:
```
> echo 'Int64.to_int 5999999999999999999L;;' | ocaml
# - : int = -3223372036854775809
```

Reply:
```
> echo 'Int64.(min_int, to_int min_int, max_int, to_int max_int);;' | ocaml
# - : int64 * int * int64 * int = (-9223372036854775808L, 0, 9223372036854775807L, -1)
```

Reply:
Ah yes of course, good catch. Do you think it would be enough to add some safeguards that fail in those situation ? Otherwise I'll have to replace every `int` occurrence with `Int63.t` (or someting). I suspect that the issue should never rise anyway :
- For `Field.Address`, reaching the `2**31` mark would mean the `b.tree` file is 2TB
- For `Field.Pointer` every manipulated value should be smaller than `Params.page_sz` which is an `int` (and also should be in the order of 4KB anyway).

Reply:
I would add the safeguards, we're never too careful.

You should use int63 everywhere you use int64, especially if it makes these int overflow go away.